spacy/gold.pyx

```diff
@@ -673,9 +694,58 @@ cdef class TokenAnnotation:
                 "heads": self.heads,
                 "deps": self.deps,
                 "entities": self.entities,
-                "morphology": self.morphology,
+                "morphs": self.morphs,
+                "sent_starts": self.sent_starts,
                 "brackets": self.brackets}
 
+    def get_id(self, i):
+        try:
+            return self.ids[i]
+        except IndexError:
+            return i
+
+    def get_word(self, i):
+        try:
+            return self.words[i]
+        except IndexError:
+            return ""
+
+    def get_tag(self, i):
+        try:
+            return self.tags[i]
+        except IndexError:
+            return "-"
+
+    def get_head(self, i):
+        try:
+            return self.heads[i]
+        except IndexError:
+            return i
+
+    def get_dep(self, i):
+        try:
+            return self.deps[i]
+        except IndexError:
+            return ""
+
+    def get_entity(self, i):
+        try:
+            return self.entities[i]
+        except IndexError:
+            return "-"
+
+    def get_morph(self, i):
+        try:
+            return self.morphs[i]
+        except IndexError:
+            return {}
+
+    def get_sent_start(self, i):
+        try:
+            return self.sent_starts[i]
+        except IndexError:
+            return None
+
```

Comment:
Many people in Python argue for this use of try/except, but personally I strongly disprefer it. i think try/except shouldn't be used for control flow, it should only be used for error handling. So I'd like to see these replaced by conditionals, probably just using the inline ternary, e.g.

```
return self.sent_starts[i] if i < len(self.sent_starts) else None
```